sklearn/metrics/_classification.py

```diff
@@ -2567,9 +2569,18 @@ def log_loss(
     ...          [[.1, .9], [.9, .1], [.8, .2], [.35, .65]])
     0.21616...
     """
+    if eps == "auto":
+        if "numpy" in str(type(y_pred)) and "float" in str(y_pred.dtype):
+            eps = np.finfo(y_pred.dtype).eps
+        else:
+            warnings.warn(
+                "eps set to 1e-15 as y_pred is not a numpy float array",
+                DataConversionWarning,
+            )
+            eps = 1e-15
+
     y_pred = check_array(y_pred, ensure_2d=False)
```

Comment:
```suggestion
    y_pred = check_array(y_pred, ensure_2d=False)
    eps = np.finfo(y_pred.dtype).eps if eps == "auto" else eps
```

Reply:
Wouldn't the line 2583 not throw an error if y_pred is a python list or a numpy integer array?

Reply:
`check_array` will convert it to a NumPy array. This is the reason why I put this line after the `check_array`.

Reply:
`check_array` converts it to a numpy array. But np.finfo would not work if it is an integer array. How should that be handled?

Reply:
This can be fixed by setting the `dtype` parameter for `check_array`, e.g. `dtype=[np.float64, np.float32]`